.octopus/deployment_process.ocl

```diff
@@ -0,0 +1,192 @@
+                $nugetExe = Join-Path -Path $nugetPackagePath -ChildPath "Tools\nuget.exe"
+                $extractedPath = $OctopusParameters["Octopus.Action.Package[cli].ExtractedPath"]
+                $packagePath = Get-ChildItem -Path $extractedPath -Name "octopus-cli*.nupkg"
+
+                $maxAttempts = 5
+                $currentAttempt = 1
+
+                while ($currentAttempt -le $maxAttempts)
+                {
+                    write-host "Pushing package ($packagePath) to Chocolatey (attempt $currentAttempt)"
+                    . $nugetExe push "$packagePath" -source $OctopusParameters["Publish:Chocolatey:FeedUrl"] -apikey $OctopusParameters["Publish:Chocolatey:ApiKey"] -NonInteractive -Verbosity detailed
+                    if ($LastExitCode -eq 0) {
+                        exit 0
+                    } else {
+                        write-host "Failed to push package - nuget returned $LastExitCode"
+                        $currentAttempt = $currentAttempt + 1
+                    }
+                }
+                exit 1
+            EOT
+            Octopus.Action.Script.ScriptSource = "Inline"
+            Octopus.Action.Script.Syntax = "PowerShell"
+        }
+        worker_pool = "hosted-windows-2019"
+
+        packages "cli" {
+            acquisition_location = "Server"
+            feed = "octopus-server-built-in"
+            package_id = "octopus-cli"
+            properties = {
+                Extract = "True"
+            }
+        }
+
+        packages "NuGet.CommandLine" {
+            acquisition_location = "Server"
+            feed = "nuget"
+            package_id = "NuGet.CommandLine"
+            properties = {
+                Extract = "True"
+            }
+        }
+    }
+}
+
+step "create-pull-request-to-update-formula-in-homebrew" {
+    name = "Create pull request to update formula in Homebrew"
+
+    action {
+        action_type = "Octopus.Script"
+        environments = ["production"]
+        properties = {
+            Octopus.Action.Script.ScriptBody = <<-EOT
+                packageVersion="$(get_octopusvariable 'Octopus.Action.Package[cli].PackageVersion')"
+                extractedPath="$(get_octopusvariable 'Octopus.Action.Package[cli].ExtractedPath')"
+                
+                username="$(get_octopusvariable 'Publish:HomeBrew:Username')"
+                email="$(get_octopusvariable 'Publish:HomeBrew:UserEmail')"
+                personalAccessToken="$(get_octopusvariable 'Publish:HomeBrew:ApiKey')"
+
+                orgName="OctopusDeploy"
+                repoName="$(get_octopusvariable 'Publish:HomeBrew:RepoName')"
+                releaseAssets=$(curl -X GET https://api.github.com/repos/$orgName/$repoName/releases | jq "map(select(.name == \"$packageVersion\"))[0].assets")
+                
+                # https://github.com/Homebrew/install/#install-homebrew-on-macos-or-linux

```

Comment:
sunk cost means it's too late to do this now, but in future because Homebrew depends on ruby, we're guaranteed to have ruby available to us so we could use ruby to do all the work, rather than writing a pile of hard to read `curl | jq` :shrug:

Reply:
I agree that all the `jq` calls makes it yuck, so I simplified it to just string "interpolation" as the URL is well-known